packages/core/admin/ee/server/services/review-workflows/stages.js

```diff
@@ -30,6 +31,17 @@ module.exports = ({ strapi }) => {
 
     async createMany(stagesList, { fields }) {
       const params = { select: fields };
+      const groupByWorkflow = (workflows, stage) => {
+        if (!workflows[stage.workflowId]) {
+          workflows[stage.workflowId] = [];
+        }
+        workflows[stage.workflowId].push(stage);
+        return workflows;
+      };
+      // As we can create several stages, we need to make sure that we don't exceed the licence threshold
+      Object.entries(stagesList.reduce(groupByWorkflow, {})).forEach((workflowStages) => {
+        workflowsValidationService.validateWorkflowStages(workflowStages);
+      });
```

Comment:
arent all the stages from the list using the same workflow always?

Reply:
In this function, nothing is preventing developer from adding stages to any workflow 🤔

Reply:
It would seem to me as a missuse of the method, but still !

This is my shining opportunity to propose a fp approach ✨ 

```js
const groupByWorkflow = (stage) => stage.workflowId;

pipe(
  groupBy(groupByWorkflow),
  forEach(workflowsValidationService.validateWorkflowStages)
)(stagesList);
```

Still, I think this only will validate the amount of stages the user is creating, but not how many stages the workflow has. So I don't see much value in it.